src/libltfs/arch/ltfs_arch_ops.h

```diff
@@ -0,0 +1,220 @@
+**
+**  THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS ``AS IS''
+**  AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+**  IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+**  ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+**  LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+**  CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+**  SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+**  INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+**  CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+**  ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+**  POSSIBILITY OF SUCH DAMAGE.
+**
+**
+**  OO_Copyright_END
+*************************************************************************************
+**
+** Module Name:
+**
+**  arch/ltfs_arch_ops.h
+**
+** Abstract:
+**
+**  LTFS specific platform methods version implementation
+**
+** Author(s):
+**
+**  Gustavo Padilla Valdez
+**  IBM Mexico Software Laboratory
+**  gustavo.padilla@ibm.com
+**
+** Revision History:
+**
+**  04/23/2025      --- GPV ---  ltfs_arch_ops created
+*************************************************************************************
+*/
+
+
+#ifndef __LTFS_ARCH_OPS_H__
+#define __LTFS_ARCH_OPS_H__
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+#include <stdarg.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <time.h>
+ 
+#define arch_safe_free(memobject)                               \
+        do {                                                    \
+            if(memobject)                                       \
+            {                                                   \
+                free(memobject);                                \
+                memobject = NULL;                               \
+            }                                                   \
+        }while(0)
+
+    inline void arch_strcpy_limited(char* dest, const char* src, int count)
+    {
+        int i;
+        for (i = 0; i < (count) && (src)[i] != '\0'; i++) {
+
+            (dest)[i] = (src)[i];
+        }
+        if (i < (count)) {
+
+            (dest)[i] = '\0';
+        }
+    }
+
+
+
+#ifdef _MSC_VER
+#include <libxml/xmlmemory.h>
+#include <corecrt_io.h>
+#include <process.h>
+#include <string.h>
+#include <unicode/umachine.h>
+
+#define SHARE_FLAG_DENYNO  _SH_DENYNO
+#define SHARE_FLAG_DENYWR   _SH_DENYWR
+#define SHARE_FLAG_DENYRD   _SH_DENYRD
+#define SHARE_FLAG_DENYRW   _SH_DENYRW
+#define PERMISSION_READWRITE   _S_IREAD | _S_IWRITE
+#define PERMISSION_READ  _S_IREAD
+#define PERMISSION_WRITE  _S_IWRITE
+#define S_IFLNK 0xA000 
+#define O_NONBLOCK 0
+#define INVALID_KEY UINT_MAX
+
+
+    #define arch_vsprintf(buffer,bufferCount, fmt, ...) vsprintf_s((buffer), (bufferCount), (fmt), __VA_ARGS__)
```

Comment:
Please do not use camel case `bufferCount`.

This is macro definition, so long variable name is not needed.